File: shop/shop/src/main/java/com/onlinebookshop/shop/service/BookService.java
package com.onlinebookshop.shop.service;

import java.util.List;

import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.GetMapping;

import com.onlinebookshop.shop.model.Author;
import com.onlinebookshop.shop.model.Book;
import com.onlinebookshop.shop.repository.BookRepository;

@Service
public class BookService {
	
	private BookRepository bookRp;

	public BookService(BookRepository book) {
		super();
		this.bookRp = book;
	}
	
	//add new book
	public void addBook(Book book)
	{
		bookRp.save(book);		
	}
	
	//fetch all books
	public List<Book> getAllBooks()
	{
		return bookRp.listBooks();
	}
	
	//fetch by id
	public Book getBookById(int id)
	{
		return bookRp.findBookById(id);
	}
	
	//update book by id
	public int updatebook(Book book)
	{
		return bookRp.updateBook(book);
	}
	
	//delete authors service
	public int deleteBook(int id)
	{
		return bookRp.deleteBook(id);
	}
	

}

File: online-book-shop/src/services/BookService.js
import axios from "axios";
const API_URL="http://localhost:2020/api/books";
class BookService {
    //fetch all books
    getAllBooks() {
        return axios.get(API_URL);
    }
    //fetch book by id
    getBookById(id){
        return axios.get(`${API_URL}/${id}`);
    }
    //add new book
    addBook(book){
        return axios.post(API_URL,book);
    }
    //delete a book by id
    deleteBook(id){
        return axios.delete(`${API_URL}/${id}`);
    }
    //update book by id
    updateBook(id,book){
        return axios.put(`${API_URL}/${id}`,book);
    }
}
export default new BookService();
 